src/components/Table/table-cells/TableBodyCell.tsx

```diff
@@ -15,16 +14,67 @@ export const TableBodyCell: React.FC<TableCellProps> = ({
     className,
     iconStart,
     iconEnd,
-    isFirstColumn,
-    canExpand,
-    isExpanded,
-    depth,
-    getToggleRowExpandedProps,
+    expandableColumnIndex,
+    row,
+    cell,
     children,
     ...props
 }) => {
-    const paddingLeft = `${depth * 0.75}rem`;
-    const marginLeft = spacings.spacingMedium;
+    // Fix this type definition of any
+    const isExpandableColumn = (row as any)?.cells[expandableColumnIndex].value === cell?.value;
+    const leftMarginBuffer = 20; // px
+    const leftMarginIndentDepth = row?.depth * 20; // px
+
+    const renderIconTextIcon = () => (
+        <IconTextIcon
+            component="div"
+            iconStart={iconStart}
+            typographyProps={{isNowrap: true}}
+        >
+            {children}
+        </IconTextIcon>
+    );
+
+    const renderTableCell = () => {
+        // These are cells that are in the expandable row (canExpand) and it is the column in
+        // which the cells show the chevron icon to expand and collapse sub-rows (isExpandableColumn)
+        if (isExpandableColumn && row?.canExpand) {
+            return (
+                <TableCell
+                    {...
+                        (row?.canExpand && isExpandableColumn
+                            ? row?.getToggleRowExpandedProps({style: {marginLeft: `${leftMarginIndentDepth}px`}})
+                            : {}
+                        )
```

Comment:
As you already check this condition above with `if (isExpandableColumn && row?.canExpand)` this condition seems unnecessary.

Reply:
Hmm, not sure if that can be removed. It's the case where the cell is in the expandable column (so it needs proper indentation), but the row itself doesn't have any nested sub-rows.

The condition after that is for all the other table cells - the ones that aren't in an expandable column.

Reply:
Removed the redundant conditional check.